### ISSUE TITLE
Cognitive contrast & paradigm ID: 'None' different from None

### ISSUE THREAD

**Issue by bcipolli** [NeuroVault/NeuroVault#395]
Expected:
- `None` to be `None`
- Fields to be available on every value.

Current behavior:
For these fields, `None` is actually the string `'None'`. In addition, sometimes the field is not present.

Here's what I see when I look at unique values (and use `None` if the field is not present in the metadata)

![image](https://cloud.githubusercontent.com/assets/4072455/12007803/1218f24e-abca-11e5-86a0-7e803b5e2029.png)

![image](https://cloud.githubusercontent.com/assets/4072455/12007800/0a67009a-abca-11e5-8faf-99aa8341deb2.png)

**Comment by bcipolli** [comment]
On second look: sometimes the value is `None`, sometimes `'None'`, and sometimes absent.

**Comment by chrisfilo** [comment]
None corresponds to unset value, 'None' corresponds to user selecting None/Other from the dropdown list (indicating that none of the paradigm in cognitive atlas are suitable). Maybe we can choose a different label for it, like "other". What do you think?

Could you give me an example of image that has absent cognitive_atlas_id field?

**Comment by bcipolli** [comment]
I see 1088 of them in my set of ~6000 images. Here's the first one:
http://neurovault.org/images/5/

**Comment by bcipolli** [comment]
Yes, 'other' would be better. Some fields already use 'other' (e.g. modality)

**Comment by chrisfilo** [comment]
But this image has cognitive_atlas field set... see http://neurovault.org/api/images/5/

**Comment by bcipolli** [comment]
You are correct, sir. The `None` comes from the value, not by its absence. I forgot that I changed it to output a dummy value, not `None`, when I was investigating.

I also mistyped (ok, copy/pasted from your comment haha) the field name as `cognitive_atlas_id`. When I fixed the field name, it's clear: neither field is absent

**Comment by bcipolli** [comment]
On second thought: 
http://neurovault.org/api/images/13099/

**Comment by chrisfilo** [comment]
This is a map coming from NIDM Result  package (hence the image_type "NIDM results statistic map"). NIDM Results does not include links to cognitive atlas at the moment hence this field is not set for such maps.

**Comment by bcipolli** [comment]
I understand. As an end-user, I don't care where it comes from; it just makes m code messier to have to deal with lots of different defaults and existence. 

I tend to use `(dict.get('value') or '').lower()` for strings, to handle non-existence, `None`, and non-uniformity of values. Alternately, I could fake it and insert the missing fields with `None`. But it'd be great to be a bit more uniform with the output.

Thanks!

**Comment by chrisfilo** [comment]
I know that uniformity can be useful, but we have different types of images
that only share affine fields. For example atlases have an extra field for
the xml with labels. They also don't have cognitive atlas tags because that
would not make much sense. We also have a system of custom user define
fields that will obviously change from collection to collection.

I can add fake fields to maintain uniformity, but they will be more or less
meaningless. Some lack of uniformity will also be inevitable. Let me know
what do you think.

**Comment by bcipolli** [comment]
I see. Seems too big a change to, say, subcategorize those into an 'extra' field that can contain (or not) any number of optional properties.

I think the best thing would be to update the API docs to indicate which fields are guaranteed to return, and mention where documentation lives for accessing some of these custom fields. Right now, the docs are pretty sparse on what to expect in return:
http://neurovault.org/api-docs
